### ISSUE TITLE
QC PACKING - can't edit

### ISSUE THREAD

**Issue by litzvi** [litzvi/avc-beta#597]
Can't be edited.

**Comment by litzvi** [comment]
[localhost-1632282310611.log](https://github.com/litzvi/avc-beta/files/7207563/localhost-1632282310611.log)

**Comment by mei309** [comment]
public void setPoWeights(TransactionProcess<?> process) {
		removeOldWeightedPos(process.getId());
		List<ItemAmountWithPoCode> poWeights = getProcessRepository().findTransactionWeightedPos(process.getId(), ItemGroup.PRODUCT);
		addPoWeights(poWeights, process, true);		
	}

**Comment by mei309** [comment]
this dose not work for qcPacking because its not product

**Comment by litzvi** [comment]
If you do it regardless of pos.
meaning treat the used items as Material used. e.g. Sugar, salt etc.

1. Easier to pack, no need to choose the pos in advance, just pack qc hold (or floor etc.) - 
2. I think is the way they do it.

Might cause some errors in report, I'll fix them as they accure.

**Comment by mei309** [comment]
i still dont think this is a good idea since we have a lot of pos with qchold and it will be all the time therefor it will be very hard to it

**Comment by mei309** [comment]
like loading makes sense but then also the packing is ok

**Comment by litzvi** [comment]
> i still dont think this is a good idea since we have a lot of pos with qchold and it will be all the time therefor it will be very hard for them to do it

correct about the problem - we need to clean up the old ones or find a way to search through it.
perhaps we can show only for a range of dates?

**Comment by litzvi** [comment]
> like loading makes sense but then also the packing way is ok especially to get their review, therefor we only need to fix the function i put

the numbers don't add up. seems like a bunch of pos add up to a few bags of QC. should ask Ahron/DAVID TOMORROW.
They also don't show the pos for QC in the reports.

**Comment by mei309** [comment]
the point of the program is that they can track everything therefor i do think like loading or packing is suitable but i will wait for their answer

**Comment by litzvi** [comment]
you are correct. he wants by po

**Comment by litzvi** [comment]
So it's by po. I fixed to get also ItemGroup.QC.
He also said they only use roasted stuff not the waste from 'cleaning'

**Comment by litzvi** [comment]
He also said they only use roasted stuff not the waste from 'cleaning'.
We need to solve the dropdown shouldn't show all old data -I'll open a separate issue for it. (issue 601)

**Comment by litzvi** [comment]
> He also said they only use roasted stuff not the waste from 'cleaning'.

Only QC hold and only from roasting and packing .

**Comment by mei309** [comment]
> So it's by po. I fixed to get also ItemGroup.QC. - STILL CAN'T EDIT SOMTIMES (ATTACHED)
> [localhost-1632389212731.log](https://github.com/litzvi/avc-beta/files/7216370/localhost-1632389212731.log)

the itemGroup is WASTE and not QC sorry

**Comment by mei309** [comment]
i think its better if we dont treat qc hold as waste then i would not need to hard code the item id in the functions below but for now can you create below function?
objectTableReader.findAvailableInventoryPoCodes(new ProductionFunctionality[]{ProductionFunctionality.ROASTER, ProductionFunctionality.PACKING}, ItemGroup.WASTE, itemId);

**Comment by mei309** [comment]
and in raw qc i think we should change the storage to waste amounts like all processes

**Comment by mei309** [comment]
only after the changes i will succeed in checking if it actually works

**Comment by litzvi** [comment]
> > So it's by po. I fixed to get also ItemGroup.QC. - STILL CAN'T EDIT SOMTIMES (ATTACHED)
> > [localhost-1632389212731.log](https://github.com/litzvi/avc-beta/files/7216370/localhost-1632389212731.log)
> 
> the itemGroup is WASTE and not QC sorry

changed

**Comment by litzvi** [comment]
> i think its better if we dont treat qc hold as waste then i would not need to hard code the item id in the functions below but for now can you create below function?
> objectTableReader.findAvailableInventoryPoCodes(new ProductionFunctionality[]{ProductionFunctionality.ROASTER, ProductionFunctionality.PACKING}, ItemGroup.WASTE, itemId);

com.avc.mis.beta.service.WarehouseManagement.findAvailableInventoryPoCodes(ProductionFunctionality[], ItemGroup, Integer)

p.a. all these functions are in WarehouseManagement

**Comment by litzvi** [comment]
you are correct, hard code the item is a wrong.

We can treat QC hold as 'QC'.
All we need is to add it to in qc items (instead of waste).
You allow them to do it in the process (btw if you do that they can pack QC directly in the process, if they happen to do that).

**Comment by litzvi** [comment]
> and in raw qc i think we should change the storage to waste amounts like all processes

don't understand

**Comment by mei309** [comment]
> you are correct, hard code the item is a wrong.
> 
> We can treat QC hold as 'QC'.
> All we need is to add it to in qc items (instead of waste).
> You allow them to do it in the process (btw if you do that they can pack QC directly in the process, if they happen to do that).

it wouldn't help because then it will give all qc items so i will still need to hard code it

**Comment by mei309** [comment]
maybe having like wasteQc group will be better

**Comment by mei309** [comment]
and its working now and i added a checkbox to choose if we want to pack all qc waste and not only roasted

**Comment by litzvi** [comment]
> > you are correct, hard code the item is a wrong.
> > We can treat QC hold as 'QC'.
> > All we need is to add it to in qc items (instead of waste).
> > You allow them to do it in the process (btw if you do that they can pack QC directly in the process, if they happen to do that).
> 
> it wouldn't help because then it will give all qc items so i will still need to hard code it

all qc? where?
you mean in 'qc packing'? you can control by production functionality. anyway, since they are choosing by po it won't be overwhelmed.

**Comment by litzvi** [comment]
> maybe having like wasteQc group will be better

Make a group for one scenario? ugly and dangerous.
imo better to hard code the item.

**Comment by litzvi** [comment]
> and its working now and i added a checkbox to choose if we want to pack all qc waste and not only roasted

Ahron said you don't need.

**Comment by mei309** [comment]
> > and its working now and i added a checkbox to choose if we want to pack all qc waste and not only roasted
> 
> from roasting and packing process
> Ahron said you don't need.

it will be a request a little later. but anyway i dont like not giving the option so i did it like in packing, if it will disturb them i will remove it

**Comment by mei309** [comment]
and i hard coded the item for now

**Comment by litzvi** [comment]
> > > and its working now and i added a checkbox to choose if we want to pack all qc waste and not only roasted
> > 
> > 
> > from roasting and packing process
> > Ahron said you don't need.
> 
> it will be a request a little later. but anyway i dont like not giving the option so i did it like in packing, if it will disturb them i will remove it

in front end i think it's good, but up to you.
